File: boilerplate/java/order-service/src/main/java/com/example/orderservice/application/usecases/SoftDeleteOrderUseCaseImpl.java
package com.example.orderservice.application.usecases;

import com.example.orderservice.domain.models.Order;
import com.example.orderservice.domain.models.OrderId;
import com.example.orderservice.domain.models.OrderNotFoundException;
import com.example.orderservice.domain.ports.OrderRepository;

public class SoftDeleteOrderUseCaseImpl {
    private final OrderRepository orderRepository;

    public SoftDeleteOrderUseCaseImpl(OrderRepository orderRepository) {
        this.orderRepository = orderRepository;
    }

    public void execute(OrderId id) {
        Order order = orderRepository.findById(id)
            .orElseThrow(() -> new OrderNotFoundException("Order not found: " + id.getValue()));
        order.softDelete();
        orderRepository.save(order);
    }
}

File: boilerplate/nestjs/order-service/src/application/usecases/soft-delete-order.use-case.impl.ts
import { Injectable, Inject } from "@nestjs/common";

import { ISoftDeleteOrderUseCase } from "./soft-delete-order.use-case.interface";
import { DomainException } from "../../domain/exceptions/domain.exception";
import { OrderId } from "../../domain/models/order-id.value-object";
import { OrderRepositoryPort } from "../../domain/ports/order-repository.port";
import { SoftDeleteOrderCommand } from "../dtos/user-order.dto";

@Injectable()
export class SoftDeleteOrderUseCaseImpl implements ISoftDeleteOrderUseCase {
  constructor(
    @Inject("OrderRepositoryPort")
    private readonly orderRepository: OrderRepositoryPort,
  ) {}

  async execute(command: SoftDeleteOrderCommand): Promise<void> {
    const id = new OrderId(command.orderId);
    const order = await this.orderRepository.findById(id);

    if (!order) {
      throw new DomainException(`Order not found: ${id.value}`);
    }

    const deletedOrder = order.softDelete();
    await this.orderRepository.save(deletedOrder);
  }
}
